### PR TITLE
Component types

### PR DESCRIPTION
**Pull request by ap-justin**
Ticket(s):
- N/A 
- bug: CountrySelector sets `isDirty:true` on mount - related to #2467 

## Explanation of the solution
* see comments for reasons of changes

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/components/CountrySelector/CountrySelector.tsx [hunk 1]
```diff
@@ -49,21 +44,6 @@ export default function CountrySelector<
 
   const [query, setQuery] = useState(country.name);
 
-  /**
```
Comment (ap-justin) [inline src/components/CountrySelector/CountrySelector.tsx:52]: removing this `useEffect`  as it makes the form it belongs to `dirty` (related to #2467 ), Forms using `<CountrySelector/>` should now take care of setting flags for persistent country records
see changes: 
EditProfile
Documentation

File: src/types/components.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { FileObject } from "./aws";
```
Comment (ap-justin) [inline src/types/components.ts:1]: as widely used component related type tend to be single for reach component, just group them in this file.

File: src/pages/Admin/Charity/Banking/RecipientDetails/types.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { CreateRecipientRequest } from "types/aws";
-import { Country } from "types/countries";
-import { OptionType } from "types/utils";
-import { Asset } from "components/registration";
+import { OptionType } from "types/components";
+import { Country } from "types/components";
+import { Asset } from "components/FileDropzone";
```
Comment (ap-justin) [inline src/pages/Admin/Charity/Banking/RecipientDetails/types.ts:4]: moved Filedropzone from `components/registration` to `components/` as it is used also in admin now

File: src/pages/Admin/Charity/Banking/RecipientDetails/getDefaultValues.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { FormValues } from "./types";
 import { AccountRequirements } from "types/aws";
-import { Country } from "types/countries";
-import { genFileAsset } from "pages/Registration/Steps/getRegistrationState";
```
Comment (ap-justin) [inline src/pages/Admin/Charity/Banking/RecipientDetails/getDefaultValues.ts:4]: moved this helper to `components/Filedropzone` as it now both used by admin and registration

File: src/components/CountrySelector/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Country } from "types/components";
+import countries from "assets/countries/all.json";
+
+export const country = (name: string): Country => {
```
Comment (misicnenad) [inline src/components/CountrySelector/helpers.ts:4]:
```suggestion
export const country = (name = ""): Country => {
```
nit: makes it unnecessary to do `country(props.hq_country ?? "")`

File: src/types/components.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+//selector
+export type ValKey = string | number;
+export type OptionType<V> = { label: string; value: V };
+
+//dropzone
+export type Asset = {
```
Comment (misicnenad) [inline src/types/components.ts:8]: The more I'm looking at it, the more I get the feeling of how generic of a name `Asset` is. Maybe a better one would be `FileDropzoneAsset`?
It'd be a small change, but would go a long way of making it obvious what its purpose is.

**Review by misicnenad — APPROVED** [review]
:+1:

File: src/types/components.ts (hunk 6 above)
Comment (ap-justin) [reply src/types/components.ts:8]: I agree with this ser, I'll include this change in #2467

**Pull request merged**